cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -443,6 +443,7 @@ class TestS3FS : public S3TestMixin {
     // Most tests will create buckets
     options_.allow_bucket_creation = true;
     options_.allow_bucket_deletion = true;
+    options_.sse_customer_key = "1WH9aTJ0+Tn0NLbTMHZn9aCW3Li3ViAdBsoIldPCREw=";
```

Comment:
Why is this needed?

Reply:
I tend to cover the workflow with sse_customer_key nonempty.

Reply:
If we always set `sse_custom_key`, we can't test no `sse_custom_key` case.
Can we add a test that uses `sse_custom_key` instead?

Reply:
Indeed, and ideally the SSE-C encryption test would write a file with the encryption key, and check that trying to read it with another key (or with no key at all) fails.